torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,540 @@
+            indices = self.labels["rsFile"] == base_path
+            ids = self.labels[indices]["id"].tolist()
+            geoms = [self.geometries[i]["geometry"]["coordinates"][0][:4] for i in ids]
+        # Test set - Task 2 has no mapping csv. Mapping is inside of geometry
+        else:
+            ids = [
+                k
+                for k, v in self.geometries.items()
+                if v["properties"]["plotID"] == base_path
+            ]
+            geoms = [self.geometries[i]["geometry"]["coordinates"][0][:4] for i in ids]
+
+        # Convert to pixel coords
+        boxes = []
+        with rasterio.open(path) as f:
+            for geom in geoms:
+                coords = [f.index(x, y) for x, y in geom]
+                xmin = min([coord[0] for coord in coords])
+                xmax = max([coord[0] for coord in coords])
+                ymin = min([coord[1] for coord in coords])
+                ymax = max([coord[1] for coord in coords])
+                boxes.append([xmin, ymin, xmax, ymax])
+
+        tensor: Tensor = torch.tensor(boxes)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load target label for a single sample.
+
+        Args:
+            path: path to image
+
+        Returns:
+            the label
+        """
+        # Find indices for objects in the image
+        base_path = os.path.basename(path)
+        indices = self.labels["rsFile"] == base_path
+
+        # Load object labels
+        classes = self.labels[indices]["taxonID"].tolist()
+        labels = [self.class2idx[c] for c in classes]
+        tensor: Tensor = torch.tensor(labels)  # type: ignore[attr-defined]
+        return tensor
+
+    def _load(self, root: str) -> Tuple[List[str], Dict[int, Dict[str, Any]], Any]:
+        """Load files, geometries, and labels.
+
+        Args:
+            root: root directory
+
+        Returns:
+            the image path, geometries, and labels
+        """
+        import pandas as pd
+
+        if self.split == "train":
+            directory = os.path.join(root, self.directories[self.split][0])
+            labels: pd.DataFrame = self._load_labels(directory)
+            geoms = self._load_geometries(directory)
+        else:
+            directory = os.path.join(root, self.task)
+            if self.task == "task1":
+                geoms = None  # type: ignore[assignment]
+                labels = None
+            else:
+                geoms = self._load_geometries(directory)
+                labels = None
+
+        images = self._load_images(directory)
+
+        return images, geoms, labels
+
+    def _load_labels(self, directory: str) -> Any:
+        """Load the csv files containing the labels.
+
+        Args:
+            directory: directory containing csv files
+
+        Returns:
+            a pandas DataFrame containing the labels for each image
+        """
+        import pandas as pd
```

Comment:
How complex is this csv? I wonder if we could get away with just using `csv` instead of `pandas`. We do that in a couple of other datasets.

Reply:
The files themselves aren't complex but I have to do some cleanup and joins to the CSVs in `_load_labels` because they are messy. Not sure we want to roll our own for that.